source/slang/slang-emit-spirv.cpp

```diff
@@ -2960,24 +2961,19 @@ struct SPIRVEmitContext
         SpvExecutionMode mode = SpvExecutionModeMax;
         for (auto globalInst : referencedBuiltinIRVars)
         {
-            if (auto thisMode = isDepthOutput(globalInst))
+            auto thisMode = getDepthOutputExecutionMode(globalInst);
+            if (mode == SpvExecutionModeMax)
+                mode = thisMode;
+            else if (mode != thisMode)
             {
                 needDepthReplacingMode = true;
-                if (mode == SpvExecutionModeMax)
-                    mode = thisMode;
-                else if (mode != thisMode)
-                    mode = SpvExecutionModeDepthReplacing;
+                mode = SpvExecutionModeDepthReplacing;
                 break;
```

Comment:
Does the code here intentionally not `break;` at `if (mode == SpvExecutionModeMax)`?

Reply:
No we can't break because the goal here is to find an evidence that we need DepthReplacing. And we don't know we don't need it unless we have examined every globalInst.